win32/Manual/Program.cs

```diff
@@ -0,0 +1,300 @@
+            out double d_arc__km, out double theta_rx, out double A_a__db, out double a__km);
+
+        [DllImport("p528.dll", CallingConvention = CallingConvention.StdCall, CharSet = CharSet.Ansi, EntryPoint = "NakagamiRice")]
+        public static extern double NakagamiRice(double K, double p);
+
+        [DllImport("p528.dll", CallingConvention = CallingConvention.StdCall, CharSet = CharSet.Ansi, EntryPoint = "NakagamiRice2")]
+        public static extern double NakagamiRice2(double K, double p);
+
+        const string NEW_DATA_TABLES_DIR = "DataTables-v5";
+        const string OLD_DATA_TABLES_DIR = "DataTables-v4";
+        const string DIFF_DATA_TABLES_DIR = "DiffDataTables";
+
+        //static void CheckNRData()
+        //{
+        //    double K = -40;
+
+        //    do
+        //    {
+        //        double p = 1;
+        //        do
+        //        {
+        //            double v1 = NakagamiRice(K, p);
+        //            double v2 = NakagamiRice2(K, p);
+
+        //            if (v1 != v2)
+        //                Console.WriteLine($"DIFF! :: K = {K} :: p = {p} :: {v1} != {v2}");
+
+        //            p += 0.5;
+        //        } while (p <= 99);
+
+        //        K += 0.1;
+        //    } while (K < 21);
+
+        //    Console.WriteLine("Done.");
+        //    Console.ReadKey();
+        //}
+
+        static void Main(string[] args)
+        {
+            //CheckNRData();
+            //return;
+
+            //ExtractThayerValues();
+            //return;
+
+            double d__km = 0;
+            double h_1__meter = 100.5;
+            double h_2__meter = 20e3;
+            double f__mhz = 30e3;
+            double time_percentage = 50;
+            Result result = new Result();
+            P528(d__km, h_1__meter, h_2__meter, f__mhz, (int)Polarization.Horizontal, time_percentage, ref result);
+
+            ////using (var fs = new StreamWriter("ray_len.csv") { AutoFlush = true })
+            ////{
+            ////    fs.WriteLine("h_1,r_1,a_1");
+
+            ////    for (int h1 = 5; h1 < 10000; h1 += 10)
+            ////    {
+            ////        P528(d__km, h1, h_2__meter, f__mhz, (int)Polarization.Horizontal, 50, ref result);
+
+            ////        fs.WriteLine($"{h1},{result.r_1__meter},{result.a_1__meter}");
+            ////    }
+            ////}
+
+            //////int rtn = P528(d__km, h_1__meter, h_2__meter, f__mhz, (int)Polarization.Horizontal, time_percentage, ref result);
+
+            Console.WriteLine($"BTL: {result.A__db}");
+            Console.WriteLine($"Free Space: {result.A_fs__db}");
+            Console.WriteLine($"Absorption: {result.A_a__db}");
+            Console.ReadKey();
+            return;
+
+            GenerateDataFiles();
+            DiffDataTables();
+
+            Console.WriteLine();
+            Console.WriteLine("Done.");
+            Console.ReadKey();
```

Comment:
Unreachable code

Reply:
manual test code that will be removed before merge